### PR TITLE
dual ts2phc /dpll support for multi  card T-GM

### PR DESCRIPTION
**Pull request by aneeshkp**
This  code adds  updates  from 1 card GM  to support for multi Card GM

### CODE REVIEW

File: pkg/event/event.go [hunk 1]
```diff
@@ -369,10 +365,11 @@ func (e *EventHandler) updateGMState(cfgName string) grandMasterSyncState {
 		clockClass: e.gmSyncState[cfgName].clockClass,
 		sourceLost: e.gmSyncState[cfgName].sourceLost,
 	}
+	fmt.Printf("%#v", rGrandMasterSyncState)
```
Comment (jzding) [inline pkg/event/event.go:368]: is this log temporary?

**Review by jzding — APPROVED** [review]
/lgtm

File: pkg/event/stats.go [hunk 2]
```diff
@@ -24,3 +22,32 @@ type DataMetrics struct {
...
+	signalSource EventSource // GNSS PPS
+	sourceLost   bool
+}
+
+func (d *Data) updateState() {
+	state := PTP_UNKNOWN
+	for _, detail := range d.Details { // 2 ts2phc or 2 dpll etc
+		switch detail.State {
+		case PTP_FREERUN: // if its free ru and main state is not holdover then this is the state
+			state = detail.State
```
Comment (jzding) [inline pkg/event/stats.go:43]: if main state is holdever  and this detail.State is the last one in the slice then the main state will be updated to freerun which is not correct.

Comment (aneeshkp) [reply pkg/event/stats.go]: Added stats_test  unit test and fixed it

**Review by jzding — APPROVED** [review]
/lgtm

**Pull request merged**